ddsmt.py

```diff
@@ -56,30 +55,29 @@ def __str__ (self):
         return "[ddsmt] Error: {}".format(self.msg)
 
 
-class DDSMTCmd (Thread):
-
-    def __init__ (self, cmd, timeout, log):
-        Thread.__init__(self)
+class DDSMTCmd ():
+    def __init__(self, cmd, timeout, log):
         self.cmd = cmd
         self.timeout = timeout
         self.log = log
 
-    def run (self):
+    def run_cmd(self, is_golden = False):
         self.process = Popen (self.cmd, stdout=PIPE, stderr=PIPE)
-        self.out, self.err = self.process.communicate()
-        self.rcode = self.process.returncode
-
-    def run_cmd (self, is_golden = False):
-        self.start()
-        self.join (self.timeout)
-        if self.is_alive():
-            self.process.terminate()
-            self.join()
+        start = time.time()
+        try:
             if is_golden:
-                raise DDSMTException ("initial run timed out")
+                self.out, self.err = self.process.communicate(timeout=60)
```

Comment:
We shouldn't have a timeout of 60 seconds by default for the first run. Either the user specifies a timeout via option `-t` or there is none.